tdms/tests/unit/test_argument_parser.cpp

```diff
@@ -0,0 +1,115 @@
+/**
+ * @file test_argument_parser.cpp
+ * @brief Tests of the argument parsing and file I/O.
+ */
+#include "argument_parser.h"
+
+#include <algorithm>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+#include <catch2/catch_test_macros.hpp>
+
+using std::string;
+using std::vector;
+
+/**
+ * @brief Test that the argument parser recovers the input arguments provided.
+ */
+TEST_CASE("Test parsing help") {
+  const char *input_args[] = {"tdms", "-h"};
+  auto args = ArgumentNamespace(2, const_cast<char **>(input_args));
+  REQUIRE(args.have_flag("-h"));
+}
+
+/** Drop the first and last component of an std::vector (needs size > 2!!). */
+template<typename T>
+vector<T> drop_first_and_last(const vector<T> &v) {
+  return vector<T>(v.begin() + 1, v.end() - 1);
+}
+
+/** Convert a vector of std::strings to an array of chars */
+const char **vector_to_array(const vector<string> &vec) {
+  const char **arr = new const char *[vec.size()] {};
+  std::transform(vec.begin(), vec.end(), arr,
+                 [](const std::string &s) { return s.c_str(); });
+  return arr;
+}
+// TODO: check with Will that this doesn't shadow the implementation
```

Comment:
I am still not super in-the-lingo; but if you're worried that the executable is meant to do this conversion itself, it doesn't.